### ISSUE TITLE
Files can't be deleted from the app itself on Windows 11 Home 22H2 

### ISSUE THREAD

**Issue by danvoronov** [suchnsuch/tangent-public#9]
Version: 0.5.6

Steps to Reproduce:

1. Install the app on Windows 11 Home 22H2 for all users to C:\Program Files\Tangent
2. Set Workspace folder to Box Drive sync folder C:\Box\Apps\tangent\ and create note in \Log
3. Try to delete a file from within the app

Expected result: Files should be deleted

Actual result: Files can't be deleted, but can be created

Additional suggestions for improvements:

- Add a keyboard shortcut for creating todos in notes
- The left/right arrow keys currently close other panels instead of moving note left/right. It would be better if notes panels could be moved with arrow keys or mouse.

Let me know if you would like me to modify or expand the issue description in any way.

**Comment by taylorhadden** [comment]
Hi Dan! Thanks for that bug report. I have a few questions:

1. Does this issue persist when installed just to the local user?
2. Does this issue persist when using a folder within the user directory?
3. Are there any errors printed in the log? You can open them from the Tangent logo menu -> Help -> Show Logs
4. Are there any errors printed in the development console? You can open it with Ctrl+Shift+i

---

I hear you on the todo keyboard shortcuts! That is on my todo list :)

Not exactly what you want, but the order of panels can be set from the "Map" view by clicking and dragging from one node on the map to another in the appropriate order. I will definitely consider your suggestion.

**Comment by danvoronov** [comment]
1. I will check.
2. No, if the file is located on the disk in the user directory, the program deletes it. The issue only occurs on the synchronized Box drive because it mounts it to a folder when Windows loads, and it seems to have some strange access permissions.

3. 
`
INFO  2023-09-14 18:50:37: [file] updating and saving D:\Box\Apps\tangent_todo\Log\2023-09-14.md
INFO  2023-09-14 18:50:59: [file] updating and saving D:\Box\Apps\tangent_todo\Log\2023-09-14.md
INFO  2023-09-14 18:53:47: [file] updating and saving D:\Box\Apps\tangent_todo\Log\New Note.md
ERROR 2023-09-14 18:53:56: [messages] Failed to delete D:\Box\Apps\tangent_todo\Log\New Note.md Error: Failed to perform delete operation

ERROR 2023-09-14 18:54:04: [messages] Failed to delete D:\Box\Apps\tangent_todo\Log\New Note.md Error: Failed to perform delete operation

INFO  2023-09-14 18:54:13: [workspace-view] Saving workspace  DAN-LAPTOP_0.workspace
INFO  2023-09-14 18:54:13: [workspace] Closing workspace D:\Box\Apps\tangent_todo
`

---

I would greatly appreciate a shortcut for tasks and the ability to switch between panels. I'm transitioning to Tangent from getslash.co app to have my data stored in my own files on my online drive, but I do like Slash's approach of having panels for the week and day todos.

I'm not particularly tied to Box drive either, and it would be better to connect a Git repository with automatic pushes.

Right now, I'm quite confused about how the panels work in Tangent. They can't be moved, and they seem to disappear randomly. Also, it seems impossible to select multiple files on the Map view using Ctrl or Shift and then switch to the Chain view to see them all displayed there.

**Comment by taylorhadden** [comment]
I wonder, does launching Tangent as an administrator help in this case?

---

I can certainly understand the confusion. The panels are styled after [Andy Matuschak's notes](https://notes.andymatuschak.org/About_these_notes). The intention is that you can start on some note, quickly link to another note with `[[Wiki Links]]`, and follow those links down whatever thread of interest you have. As you keep following links, the panels stack up so that you always have full context of where you've come from.

I must admit, though, that missing Shift+Click is pretty basic! I'll definitely be adding that to my list of things to do in the current alpha cycle. The upcoming 0.6 version also features full thread history support so you can easily restore previous threads.

**Comment by danvoronov** [comment]
Running the program with administrator rights does not help

From  development console:
`
NoteEditor.svelte:597 Resuming focus to  to_delete (2) [11, 11]
MapView.svelte:638 ignore-until-reset
MapView.svelte:640 zoom-or-focus
NoteEditor.svelte:597 Resuming focus to  to_delete (2) [11, 11]
MapView.svelte:638 ignore-until-reset
MapView.svelte:640 zoom-or-focus
NoteEditor.svelte:597 Resuming focus to  to_delete (2) [11, 11]
MapView.svelte:638 ignore-until-reset
MapView.svelte:640 zoom-or-focus
NoteEditor.svelte:597 Resuming focus to  to_delete (2) [11, 11]0: 111: 11length: 2[[Prototype]]: Array(0)at: ƒ at()concat: ƒ concat()constructor: ƒ Array()copyWithin: ƒ copyWithin()entries: ƒ entries()every: ƒ every()fill: ƒ fill()filter: ƒ filter()find: ƒ find()findIndex: ƒ findIndex()findLast: ƒ findLast()findLastIndex: ƒ findLastIndex()flat: ƒ flat()flatMap: ƒ flatMap()forEach: ƒ forEach()includes: ƒ includes()indexOf: ƒ indexOf()join: ƒ join()keys: ƒ keys()lastIndexOf: ƒ lastIndexOf()length: 0map: ƒ map()pop: ƒ pop()push: ƒ push()reduce: ƒ reduce()reduceRight: ƒ reduceRight()reverse: ƒ reverse()shift: ƒ shift()slice: ƒ slice()some: ƒ some()sort: ƒ sort()splice: ƒ splice()toLocaleString: ƒ toLocaleString()toReversed: ƒ toReversed()toSorted: ƒ toSorted()toSpliced: ƒ toSpliced()toString: ƒ toString()unshift: ƒ unshift()values: ƒ values()with: ƒ with()Symbol(Symbol.iterator): ƒ values()Symbol(Symbol.unscopables): {at: true, copyWithin: true, entries: true, fill: true, find: true, …}[[Prototype]]: Object
App.svelte:215 Refocusing to undefined
App.svelte:215 Refocusing to undefined
App.svelte:215 Refocusing to undefined
App.svelte:215 Refocusing to undefined
ThreadView.svelte:162 clicked changing current
File.ts:96 saving to_delete
File.ts:96 saving to_delete
App.svelte:215 Refocusing to undefined
Workspace.ts:180 Firing context command: _v {canExecuteDirty: false, subscribers: Array(1), workspace: Ny}canExecuteDirty: falsesubscribers: Array(1)0: ()=>{t.postMenuUpdate({[e]:this.translateCommandToMenu(n)})}length: 0name: ""arguments: (...)caller: (...)[[FunctionLocation]]: Workspace.ts:143[[Prototype]]: ƒ ()[[Scopes]]: Scopes[5]length: 1[[Prototype]]: Array(0)workspace: Ny {creationRules: o, debug: {…}, api: {…}, dataTypes: Array(5), nodeConstructor: ƒ, …}[[Prototype]]: jmconstructor: class _vexecute: ƒ execute(e)getLabel: ƒ getLabel(e)[[Prototype]]: Bh
ThreadView.svelte:162 clicked changing current
App.svelte:215 Refocusing to undefined
App.svelte:215 Refocusing to undefined
`


---
There was also a problem with displaying images created in Obsidian like ![Untitled](assets/22/Untitled%2022.png)

**Comment by taylorhadden** [comment]
Can you split out that image element to a new issue?

**Comment by taylorhadden** [comment]
I assume that other editors are not having problems here.

Just to make sure the bases are covered:
1. Tangent can create new files in these paths?
2. Tangent can edit files in these paths?

**Comment by danvoronov** [comment]
Yes Tangent can create new files and edit files names and path. The issue only occurs with deletion. Фтв there are no such problems in Obsidian or LogSec.

**Comment by taylorhadden** [comment]
Were you able to determine what about the access permissions are different? I don't have a Box account.

**Comment by danvoronov** [comment]
I have Box account, but I don't have the deep technical knowledge to tinker with the Windows OS and understand how a closed-source program mounts itself within the file system. You can easily register Box basic account - it's free.

**Comment by taylorhadden** [comment]
There is a new debug option that allows you to directly delete files instead of sending them to the system trash / recycle bin. Does that fix this issue?
